### ISSUE TITLE
Ethnicity - UK SNOMED Categories

### ISSUE THREAD

**Issue by brianmackenna** [opensafely/codelist-development#126]
I have drafted a new [SNOMED UK category codelists here](https://www.opencodelists.org/codelist/opensafely/ethnicity-uk-categories/2b48c5be/#full-list)

_Why do we need **another** ethnicity codelist?_

The current gold standard OpenSAFELY codelist for [ethnicity is here](https://www.opencodelists.org/codelist/opensafely/ethnicity/2020-04-27/#full-list). It is a `CTV3` codelist with categories for broad (6) and detailed (16) groupings. 

More recently for federated analysis across EMIS and TPP we have been using the PRIMIS ethnicity codes. These are set up to foreground 2001 in the [main codelist here](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/eth2001/v1/#full-list). This is a definite choice as the NHS standard for ethnicity in the [data dictionary remains 2001](https://www.datadictionary.nhs.uk/data_elements/ethnic_category.html?hl=ethnic) (we are unsure why). Therefore anyone using this codelist misses a fair chunk of codes e.g. any related to the 2011 standard.

PRMIS have another codelist [Any other ethnicity code](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/non_eth2001/v1/#full-list) that does include the 2011 standards but this has two issues
1. It requires someone to know that you need multiple codelists to construct ethnicty in snomed unlike in `ctv3` 
1. It does not have categories meaning people with a 2011 code are unhelpfully lumped into the "other" category

If people agree this is enough justification I will need to add categorisation to the draft list above. @inglesp I would be grateful on advice on how to do this technically for two reasons a) you seem to have done it on the main  [primis list](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/eth2001/v1/#full-list) so maybe you have code to do it? b) if I do it it will be downloaded to excel/sheets but risk is that SNOMED codes will be truncated before uploading again. (related to this issue in [OpenCodelists-core](https://github.com/opensafely-core/opencodelists/issues/507) )

**Comment by alexwalkercebm** [comment]
Definitely sounds like a more complete list is needed.

Is this new codelist a combination of the Primis 2001 and 2011 lists, or created anew? Is there any sort of cross checking we could do with the CTV3 list?

**Comment by inglesp** [comment]
@brianmackenna have you seen https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/eth2001/v1/?  It was built by combining the separate PRIMIS ethnicity codelists.

**Comment by brianmackenna** [comment]
Yes - it only has 2001 codes in there so we miss all 2011 census codes and any others, it's one of the linked codelists above.

**Comment by inglesp** [comment]
Ah yes sorry.

So to categorise, I'd suggest downloading the codelist as a CSV, importing into a Google Sheet, and categorising manually.  We can then (a) validate the 2001 categorisations against the PRIMIS eth2001 codelist, and (b) manually check them.

Have you seen there are other ethnicity codes, not under Ethnic group finding (397731000)?  For instance, there's a whole bunch under [Ethnic group](https://www.opencodelists.org/snomedct/concept/372148003/), and there are 9996 occurrences of [Afro-Caribbean](https://www.opencodelists.org/snomedct/concept/413465009/) in EMIS.

**Comment by brianmackenna** [comment]
I hadn't! Right so we need a two step process.

1. Agreed list of codes 
2. categorise the agreed list

**Comment by brianmackenna** [comment]
On 1. This is quite  a big job and because SNOMED is comprehensive there are codes classed as ethnic groupings that verge into nationality. I will convene short discussion with @rohinimathur  and others to decide best way of tackling this!

**Comment by rc16** [comment]
Hi - just coming back to this issue as I'd like to use the SNOMED ethnicity categories. 

I have a csv file with the categories from Rohini but I'm not able to add it to opencodelists. When I upload it the categories are not included. The solution I'd seen here https://github.com/opensafely/documentation/discussions/442 from Peter Inglesby doesn't work for me. When I go to the page at the start, ending with add, it just comes to an error page.

**Comment by rc16** [comment]
Hi,

Caroline has added the categories to the ethnicity codelist: https://www.opencodelists.org/codelist/opensafely/ethnicity-with-categories-snomed/5eb3345b/ 

It just needs signing off @rohinimathur

**Comment by andrewscolm** [comment]
Working with @rohinimathur and @alexwalkerepi we expanded the [SNOMED ethnicity codelist ](https://www.opencodelists.org/codelist/opensafely/ethnicity-uk-categories/516dea7b/#full-list ), created by @brianmackenna and @rohinimathur, identifying 657 ethnicity SNOMED codes. By comparing the codelist to the [ctv3 codelist](https://www.opencodelists.org/codelist/opensafely/ethnicity/2020-04-27/), the [primis codelist](https://www.opencodelists.org/codelist/primis-covid19-vacc-uptake/eth2001/v1/), and a codelist from [GDPPR ](https://digital.nhs.uk/coronavirus/gpes-data-for-pandemic-planning-and-research/guide-for-analysts-and-users-of-the-data) we grouped the codelist according to the [2001 census](https://www.ethnicity-facts-figures.service.gov.uk/style-guide/ethnic-groups#2001-census).

https://www.opencodelists.org/codelist/user/candrews/full_ethnicity_coded/6281e42c/

**Comment by andrewscolm** [comment]
The NHS classification is based on 2001 Census classifications and does not capture information on any of the GRT groups separately. This issue is discussed in https://web.archive.org/web/20220213182343/https://www.ethnicity-facts-figures.service.gov.uk/summaries/gypsy-roma-irish-traveller. Based on the gov.uk summary above we have decided to categorise Irish Travellers as White Other.

**Comment by andrewscolm** [comment]
Quite a few of the codes in the SNOMED list have descriptions that are nationalities like "English", "Czech", "Poles", "Egyptian". After discussion it was decided to take a pragmatic / consistent approach and categorise these according to region (e.g. White British, Other White, Other White, and African respectively for the above).

**Comment by andrewscolm** [comment]
Published version with categories that don't have ethnicity information (e.g. "Ethnic category - 2001 census") removed.

https://www.opencodelists.org/codelist/opensafely/ethnicity-snomed-0removed/2e641f61/
